### DISCUSSION TITLE
What prevents a blockchain snark from using a txn snark of a partially proved zkapp command?

### DISCUSSION THREAD

**Discussion by L-as** [MinaProtocol/mina#14451] [Q&A] [answered]
I'm looking at https://github.com/MinaProtocol/mina/blob/ea6d94c5343511d754e42b08c0a5e55d96edd30c/src/lib/blockchain_snark/blockchain_snark_state.ml#L273 and can't tell what prevents me from proving part of the zkapp command. I suppose it would mean that the second pass ledger isn't updated while the target local state is?

**Comment by mrmr1993** [comment, marked as answer]
Nothing. It's enforced by the consensus layer (in particular, by the scan state).
There's a proposal to include a merkle list of txns in a block's statement, but this hasn't been implemented for the Berkeley hard-fork.
